### ISSUE TITLE
ruby: Cannot start multiple Safari drivers concurrently

### ISSUE THREAD

**Issue by p0deje** [SeleniumHQ/selenium#1160]
An attempt to start multiple Safari drivers concurrently maybe result in process hanging infinitely. That is because the server created to communicate with Safari extension is bind to the same port. Related to #736

```ruby
require 'selenium-webdriver'
require 'timeout'

Timeout.timeout(10) do
  threads, drivers = [], []
  5.times do
    threads << Thread.new do
      drivers << Selenium::WebDriver.for(:safari)
    end
  end

  threads.each do |thread|
    thread.abort_on_exception = true
    thread.join
  end

  drivers.each_with_index do |driver, index|
    puts "Checking ##{index + 1}: #{driver}"
    puts "  it runs on #{driver.send(:bridge).instance_variable_get(:@server).instance_variable_get(:@port)}"

    driver.title # make any wire call
    driver.quit
  end
end
```

Running this script will timeout the execution.

**Comment by uchuugaka** [comment]
It should be a bug to be hard coded to a "default" port as a port may not be available, it should always pass zero and let the sockets stack return an appropriate available port.

HOWEVER, to do what you want to do, please look at the ruby source like i did.
../selenium/rb/lib/selenium/webdriver/safari/options.rb
(some subset of that path should be in the selenium gem as well)
Basically, if you google a bit you'll find an example, but you should be creating an options arg to pass when creating your Safari driver object. options.rb shows you what these are. Passing zero for the port will give the appropriate effect. If you need to know the port later, you'll have to ask the driver object what port it got.
(the sockets stack never guarantees you can have a port except when it is reserved by the system and you are the system)

**Comment by p0deje** [comment]
@uchuugaka This is already fixed in https://github.com/p0deje/selenium/commit/2e9e43604eccd35423d0b6fa92fea655547a8201, I've just created this bug for tracking and reference.

**Comment by uchuugaka** [comment]
Awesome. Now if the Java stack would get in order too. ...

**Comment by p0deje** [comment]
Fixed in 94a787fb54e02ca80b17ccd475255b5c7fc96f5c
